adapter/memQuota/memQuota.go

```diff
@@ -296,46 +289,65 @@ func (mq *memQuota) reapDedup() {
 	}
 }
 
-// Produce a unique key representing the cell
-func makeKey(args adapter.QuotaArgs) string {
+// Produce a unique key representing the given labels.
+func makeKey(name string, labels map[string]interface{}) string {
 	ws := keyWorkspacePool.Get().(*keyWorkspace)
 	keys := ws.keys
 	buffer := ws.buffer
 
-	// get all the label names
-	for k := range args.Labels {
+	// ensure stable order
+	for k := range labels {
 		keys = append(keys, k)
 	}
-
-	// ensure stable order
 	sort.Strings(keys)
 
-	buffer = append(buffer, []byte(args.Name)...)
+	buffer.WriteString(name)
 	for _, k := range keys {
-		buffer = append(buffer, []byte(";")...)
-		buffer = append(buffer, []byte(k)...)
-		buffer = append(buffer, []byte("=")...)
+		buffer.WriteString(";")
+		buffer.WriteString(k)
+		buffer.WriteString("=")
 
-		switch v := args.Labels[k].(type) {
+		switch v := labels[k].(type) {
```

Comment:
non-blocker: one thing we might consider here is having the switch set a string var and then having the buffer.WriteString() call in one place (so we could handle the error, etc.).

```
var valStr string
switch v := labels[k].(type) {
case string:
   valStr = v
case int64:
   valStr = strconv.FormatInt(v, 16)
...
}
if _, err := buffer.WriteString(valStr); err != nil {
   // error handling 
}
```

We'd have to use the buffer of the other Pool in `case map[string]string`, so it gets a little awkward... but just food for thought.